packages/lisk-p2p/src/peer_pool.ts

```diff
@@ -64,24 +67,33 @@ export {
 	EVENT_INBOUND_SOCKET_ERROR,
 };
 
+interface PeerPoolConfig {
+	readonly connectTimeout?: number;
+	readonly ackTimeout?: number;
+}
+
 export const MAX_PEER_LIST_BATCH_SIZE = 100;
 export const MAX_PEER_DISCOVERY_PROBE_SAMPLE_SIZE = 100;
+export const MAX_PEER_CONNECT_ATTEMPTS = 100;
```

Comment:
do you think that this number should be smaller? like 5 connect attempts or less?

According to LIP,

> If a connection attempt fails, a different peer is selected by the procedure above. Peers in the collection triedPeers are moved to the collection newPeers after 3 consecutive failed attempts (i.e., the peer is selected randomly 3 times and every time the connection failed).

Reply:
Ah I meant to remove `MAX_PEER_CONNECT_ATTEMPTS` - It's not used in the code. Right now it removed the peer from the peer pool if it fails a single attempt. It will always be able to discover later so it's fine. We should do this for the LIP phase definitely.